aws/data_source_aws_route_table_ids.go

```diff
@@ -0,0 +1,68 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsRouteTableIDs() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsRouteTableIDsRead,
+		Schema: map[string]*schema.Schema{
+
+			"tags": tagsSchemaComputed(),
+
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
According the [EC2 API Reference](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRouteTables.html), `vpc_id` should not be a required argument. Its valid to search for route tables across multiple VPCs. Can you please:
* Change the attribute to `Optional: true`
* Use `d.GetOk("vpc_id")` instead of expecting it to always exist:

```go
if v, ok := d.GetOk("vpc_id"); ok {
	req.Filters = buildEC2AttributeFilterList(
		map[string]string{
			"vpc-id": v.(string),
		},
	)
}
```

* Change `d.SetId(d.Get("vpc_id").(string))` to `d.SetId(resource.UniqueId())`
* Ensure the documentation is updated for the attribute